app/records/api.py

```diff
@@ -1,13 +1,49 @@
-from app.ciim.client import ClientAPI
+from app.lib.api import JSONAPIClient, ResourceNotFound
+from app.records.models import APIResponse, APISearchResponse, Record
 from django.conf import settings
 
 
-def get_records_client():
-    return ClientAPI(
-        base_url=settings.CLIENT_BASE_URL,
-        api_key=settings.CLIENT_KEY,
-        verify_certificates=settings.CLIENT_VERIFY_CERTIFICATES,
-    )
+def rosetta_request_handler(uri, params={}) -> dict:
+    """Prepares and initiates the api url requested and returns response data"""
+    api_url = settings.ROSETTA_API_URL
+    if not api_url:
+        raise Exception("ROSETTA_API_URL not set")
+    client = JSONAPIClient(api_url)
+    client.add_parameters(params)
+    data = client.get(uri)
+    return data
 
 
-records_client = get_records_client()
+def record_details_by_id(id, params={}) -> Record:
+    uri = "get"
+    params = params | {
+        "id": id,
+    }
+    results = rosetta_request_handler(uri, params)
+    if "data" not in results:
+        raise Exception(f"No data returned for id {id}")
+    if len(results["data"]) > 1:
+        raise Exception(f"Multiple records returned for id {id}")
+    if len(results["data"]) == 1:
+        record_data = results["data"][0]
+        response = APIResponse(record_data)
+        return response.record
+    raise ResourceNotFound(f"id {id} does not exist")
+
+
+def record_details_by_ref(reference, params={}):
+    # TODO: Implement record_details_by_ref once Rosetta has support
+    pass
+
+
+def search_records(query, params={}) -> APISearchResponse:
+    uri = "search"
+    params = params | {
+        "q": query,
+    }
```

Comment:
Although it returns records, shouldn't `search_records` be part of the `search` app?

Reply:
Maybe, but the naming of the app is just `records` rather than something more specific like `record_details` or simply `details` so searching is just a subset of records. I think the naming is not great.

We can move these about later once more work has been done on the search.